examples/Example2/example2.cu

```diff
@@ -41,8 +47,154 @@ struct Scoring {
   }
 };
 
+#include "IterationStats.h"
+#include "initTracks.h"
+
+// Statistics for propagation chords
+IterationStats     *chordIterStats     = nullptr;
+__device__
+IterationStats_dev *chordIterStats_dev = nullptr;
+
+__host__ void PrepareStatistics()
+{
+  chordIterStats = new IterationStats();
+  // chordIterStats_dev = chordIterStats->GetDevicePtr();
+  IterationStats_dev* ptrDev= chordIterStats->GetDevicePtr();
+  cudaMemcpy(&chordIterStats_dev, &ptrDev,
+              sizeof(IterationStats_dev*), cudaMemcpyHostToDevice);
+  // Add assert(chordIterStats_dev != nullptr); in first use !?
+}
+
+__host__ void ReportStatistics( IterationStats & iterStats )
+{
+  // int  maxChordItersGPU;
+  // cudaMemcpy(&maxChordItersGPU, maxItersDone_dev, sizeof(int), cudaMemcpyDeviceToHost);   
+   std::cout << "-  Chord iterations: max (dev) = " << iterStats.GetMax() // GetMaxFromDevice()
+             << "  total iters = " <<  iterStats.GetTotal() /*GetTotalFromDevice() */  ;
+   // printf(" -- host = %4d \n", GetMaxIterationsDone_host() );
+   // std::cout << std::endl; // printf(" \n");   
+}
+
+#include "ConstBzFieldStepper.h"
+
+
 constexpr double kPush = 1.e-8;
 
+// Determine the step along curved trajectory for charged particles in a field.
+//  ( Same name as as navigator method. )
+__device__  //  __host__
+double
+FieldPropagator__ComputeStepAndPropagatedState( track   & aTrack,
```

Comment:
I think we should start modularizing more the current 'main' and move such functionality to separate files/modules. Shall we create 'magneticfield' directory and have all the stuff related to it there?

Reply:
I am working on this.  My next step is to move this into a "fieldPropagatorConstBz" class, a similar "fieldPropagatorUniformB" and later one that uses Runge-Kutta.

Reply:
I agree - I am separating it already, so that I can use it in the test (like field_cu) and in multiple examples. 

I agree with the idea of a 'magneticfield' directory - and in fact have created it already.  It will be in one of my next commits - either in this MR or the next.